tests/zkevm/test_worst_memory.py

```diff
@@ -172,10 +227,36 @@ def test_worst_returndatacopy(
     helper_contract = pre.deploy_contract(code=code)
 
     # We create the contract that will be doing the RETURNDATACOPY multiple times.
-    code_prefix = Op.STATICCALL(address=helper_contract) if size > 0 else Bytecode()
+    returndata_gen = Op.STATICCALL(address=helper_contract) if size > 0 else Bytecode()
     dst = 0 if fixed_dst else Op.MOD(Op.GAS, 7)
-    attack_block = Op.RETURNDATACOPY(dst, Op.PUSH0, Op.RETURNDATASIZE)
-    code = code_loop_precompile_call(code_prefix, attack_block, fork)
+    attack_iter = Op.RETURNDATACOPY(dst, Op.PUSH0, Op.RETURNDATASIZE)
+
+    jumpdest = Op.JUMPDEST
+    jump_back = Op.JUMP(len(returndata_gen))
+    # The attack loop is constructed as:
+    # ```
+    # JUMPDEST(#)
+    # RETURNDATACOPY(...)
+    # RETURNDATACOPY(...)
+    # ...
+    # STATICCALL(address=helper_contract)
+    # JUMP(#)
+    # ```
+    # The goal is that once per (big) loop iteration, the helper contract is called to
+    # generate fresh returndata to continue calling RETURNDATACOPY.
+    max_iters_loop = (
+        max_code_size - 2 * len(returndata_gen) - len(jumpdest) - len(jump_back)
+    ) // len(attack_iter)
+    code = (
+        returndata_gen
+        + jumpdest
+        + sum([attack_iter] * max_iters_loop)
+        + returndata_gen
+        + jump_back
+    )
+    assert len(code) <= max_code_size, (
+        f"Code size {len(code)} is not equal to max code size {max_code_size}."
+    )
```

Comment:
This was a suggestiong by @jochem-brouwer. After each loop we get some fresh returned data since GAS changed, so sounds like a nice thing to do that gets amortized.